api/core/span_context.go

```diff
@@ -52,3 +63,17 @@ func (sc SpanContext) TraceIDString() string {
 	p2 := fmt.Sprintf("%.16x", sc.TraceID.Low)
 	return p1[0:3] + ".." + p2[13:16]
 }
+
+// IsSampled returns true if the span will be exported.
+func (sc SpanContext) IsSampled() bool {
+	return sc.TraceOptions.IsSampled()
+}
+
+// SetSampled	 sets the TraceOptions bit that determines whether the span will be exported.
+func (sc SpanContext) SetSampled(sampled bool) {
```

Comment:
I'm trying to imagine how this will be used?  Note this takes a _struct_, not a pointer, so this assignment won't stick.

Reply:
Right now there is only 1 bit used in the options but if there were more bits I would think it would be better to simply define a bitmap
```go
const (
	TraceOptionBitSampled  = TraceOptions(0x01)
	TraceOptionBitReserved = TraceOptions(0x10)
)

// TraceOptions contains options associated with a trace span.
type TraceOptions byte

// IsSet returns true if the span will be exported.
func (t TraceOptions) IsSet(o TraceOptions) bool {
	return t&o == o
}

type SpanContext struct {
	TraceID      TraceID
	SpanID       uint64
	TraceOptions TraceOptions
}

sc := SpanContext{
	TraceID: TraceID{
		High: uint64(42),
		Low:  uint64(42),
	},
	TraceOptions: TraceOptionBitSampled | TraceOptionBitReserved,
}

if sc.TraceOptions.IsSet(TraceOptionBitSampled) {
	//do something
}

//OR 

if sc.TraceOptions & TraceOptionBitSampled == TraceOptionBitSampled {
    //do something
}